frontend/src/hooks/usePagination.tsx

```diff
@@ -3,9 +3,16 @@ import { useState } from "react";
 import { OrderByDirection } from "@app/hooks/api/generic/types";
 import { useDebounce } from "@app/hooks/useDebounce";
 
-export const usePagination = <T extends string>(initialOrderBy: T) => {
+export const usePagination = <T extends string>(
```

Comment:
I wonder if we should support "jump to a page number" type of navigation given that we're doing this effort under the assumption that there could be 1k or 2k entries 

<img width="1271" alt="image" src="https://github.com/user-attachments/assets/9d3b5c72-877d-4b3a-97b0-564b5ef92cd4">

Reply:
I think this would be a good idea; creating a separate ticket for it!